pageserver/src/walredo.rs

```diff
@@ -515,6 +516,71 @@ impl PostgresRedoManager {
                     LittleEndian::write_u32(&mut page[memberoff..memberoff + 4], member.xid);
                 }
             }
+            ZenithWalRecord::CsnLogSetCommitted { xids, lsn } => {
+                let (slru_kind, segno, blknum) =
+                    key_to_slru_block(key).or(Err(WalRedoError::InvalidRecord))?;
+                assert_eq!(
+                    slru_kind,
+                    SlruKind::Csn,
+                    "CsnLogSetCommitted record with unexpected key {}",
+                    key
+                );
+
+                for &xid in xids {
+                    let pageno = xid as u32 / pg_constants::CSN_LOG_XACTS_PER_PAGE;
+                    let expected_segno = pageno / pg_constants::SLRU_PAGES_PER_SEGMENT;
+                    let expected_blknum = pageno % pg_constants::SLRU_PAGES_PER_SEGMENT;
+
+                    // Check that we're modifying the correct CsnLog block.
+                    assert!(
+                        segno == expected_segno,
+                        "CsnLogSetCommitted record for XID {} with unexpected key {}",
+                        xid,
+                        key
+                    );
+                    assert!(
+                        blknum == expected_blknum,
+                        "CsnLogSetCommitted record for XID {} with unexpected key {}",
+                        xid,
+                        key
+                    );
+                    transaction_id_set_csn(xid, *lsn, page);
+                }
+
+
+            }
+            ZenithWalRecord::CsnLogSetAborted { xids } => {
+                let (slru_kind, segno, blknum) =
+                    key_to_slru_block(key).or(Err(WalRedoError::InvalidRecord))?;
+                assert_eq!(
+                    slru_kind,
+                    SlruKind::Csn,
+                    "CsnLogSetAborted record with unexpected key {}",
+                    key
+                );
+
+                for &xid in xids {
+                    let pageno = xid as u32 / pg_constants::CSN_LOG_XACTS_PER_PAGE;
+                    let expected_segno = pageno / pg_constants::SLRU_PAGES_PER_SEGMENT;
+                    let expected_blknum = pageno % pg_constants::SLRU_PAGES_PER_SEGMENT;
+
+                    // Check that we're modifying the correct CSN block.
+                    assert!(
+                        segno == expected_segno,
+                        "CsnLogSetAborted record for XID {} with unexpected key {}",
+                        xid,
+                        key
+                    );
+                    assert!(
+                        blknum == expected_blknum,
+                        "CsnLogSetAborted record for XID {} with unexpected key {}",
+                        xid,
+                        key
+                    );
+
+                    transaction_id_set_csn(xid, AbortedXidCSN, page);
```

Comment:
use `pg_constant::AbortedXidCSN` so we don't need to `use`  it above

Reply:
Done.